tool-plugins/intellij/src/test/java/io/ballerina/parsing/BallerinaParsingTest.java

```diff
@@ -0,0 +1,127 @@
+/*
+ *  Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package io.ballerina.parsing;
+
+import com.intellij.openapi.fileEditor.impl.LoadTextUtil;
+import com.intellij.psi.PsiFile;
+import com.intellij.testFramework.LightVirtualFile;
+import com.intellij.testFramework.ParsingTestCase;
+import io.ballerina.plugins.idea.BallerinaParserDefinition;
+import org.jetbrains.annotations.NonNls;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static io.netty.util.internal.StringUtil.EMPTY_STRING;
+
+/**
+ * Parsing tests.
+ */
+public class BallerinaParsingTest extends ParsingTestCase {
+
+    private final String testDataPath =
+            "../../composer/modules/integration-tests/src/test/resources/ballerina" + "-examples/examples";
+    private final String expectedResultsPath = "src/test/resources/testData/parsing/BBE/expectedResults";
+
+    public BallerinaParsingTest() {
+        super("", "bal", new BallerinaParserDefinition());
+    }
+
+    @Override
+    protected String getTestDataPath() {
+        return testDataPath;
+    }
+
+    private String getExpectedResultPath() {
+        return expectedResultsPath;
+    }
+
+    @Override
+    protected boolean skipSpaces() {
+        return false;
+    }
+
+    @Override
+    protected boolean includeRanges() {
+        return true;
+    }
+
+    public void testForBBE() {
+        //This flag is used to include/filter BBE testerina files in lexer testing
+        boolean includeTests = false;
+        Path path = Paths.get(getTestDataPath());
+        doTestDirectory(path, includeTests);
+    }
+
+    private void doTestDirectory(Path path, boolean includeTests) throws RuntimeException {
+        try {
+            File resource = path.toFile();
+            if (resource.exists()) {
+                if (resource.isFile() && resource.getName().endsWith(myFileExt)) {
+                    doTest(resource, true);
```

Comment:
Remove the 2nd parameter since it is always `true`.

Reply:
Removed